### PR TITLE
CI: regularly update nix lock file

### PR DESCRIPTION
**Pull request by imincik**
Add CI job to regularly update flake.lock file. This will keep GRASS package and development environment dependencies up-to-date.

I think, this is the last piece of Nix puzzle :).

### CODE REVIEW

File: .github/workflows/update-nix.yml
Comment (echoix) [inline .github/workflows/update-nix.yml:1]: Do you mind having this as a second job in the periodic update workflow already there? I don't mind that the frequency would be increased, there isn't much activity with the existing job.

**Comment by echoix** [comment]
Also, do you have experience with https://docs.renovatebot.com/modules/manager/nix/?

We enabled precommit, that is also opt-in.

**Comment by imincik** [comment]
> Also, do you have experience with https://docs.renovatebot.com/modules/manager/nix/?
> 
> We enabled precommit, that is also opt-in.

No, I just heard that renovate will support Nix as well. Great that it is available now. I checked the list of remaining bugs and none of them are blockers. We can use renovate instead of current PR implementation.

File: .github/workflows/update-nix.yml
Comment (imincik) [reply .github/workflows/update-nix.yml:1]: I was thinking about it as well, but let's try to use Renovate.

**Comment by echoix** [comment]
So you want to change this PR or have a new one for it?

**Pull request closed without merging**